osgp/protocol-adapter-iec60870/osgp-protocol-adapter-iec60870/src/main/java/org/opensmartgridplatform/adapter/protocol/iec60870/infra/messaging/DeviceRequestMessageListener.java

```diff
@@ -96,15 +95,11 @@ private void sendNotSupportedException(
               exception);
 
       final ProtocolResponseMessage protocolResponseMessage =
-          new ProtocolResponseMessage.Builder()
-              .messageMetadata(messageMetadata)
-              .domain(messageMetadata.getDomain())
-              .domainVersion(messageMetadata.getDomainVersion())
+          ProtocolResponseMessage.newBuilder()
+              .messageMetadata(messageMetadata.builder().withScheduled(false).build())
               .result(ResponseMessageResultType.NOT_OK)
               .osgpException(osgpException)
               .dataObject(objectMessage.getObject())
-              .retryHeader(new RetryHeader())
```

Comment:
Did you remove this intentionally? (looks like it's removed in several locations)

Reply:
I did indeed remove this intentionally.
If there is a call to the builders `messageMetadata` method - as should be there in places I removed the call to `retryHeader(new RetryHeader())`, it calls `superBuilder.withMessageMetadata` which initializes the `retryHeader` with `new RetryHeader()`.
Overall I am not entirely happy with the way this `RetryHeader` is used in combination with the message metadata and response messages in getting the retries to work as intended, but improving upon it would require some more study and seemed out of scope.